src/Vgrep/Environment/Config/Sources/File.hs

```diff
@@ -322,3 +303,44 @@ styleToVty = \case
     Blink        -> Vty.blink
     Dim          -> Vty.dim
     Bold         -> Vty.bold
+
+
+instance FromJSON KeybindingsMonoid where
+    parseJSON = genericParseJSON jsonOptions
+
+instance FromJSON Command where
+    parseJSON = genericParseJSON jsonOptions
+
+instance FromJSON (Map Key.Chord Command) where
+    parseJSON = parseJSON >=> mapMKeys parseChord
+
+mapMKeys :: (Monad m, Ord k') => (k -> m k') -> Map k v -> m (Map k' v)
+mapMKeys f = fmap M.fromList . M.foldrWithKey go (pure [])
+  where
+    go k x mxs = do
+        k' <- f k
+        xs <- mxs
+        pure ((k', x) : xs)
```

Comment:
Applicative is enough,

```haskell
go k x mxs = liftA2 (\k' xs -> (k', x) : xs) 
                    (f k) 
                    mxs
```

With `-XApplicativeDo` this should even be possible using the more readable do-style. Might need a newer LTS for that though.